pycode/memilio-generation/memilio/tools/README.md

```diff
@@ -0,0 +1,81 @@
+
+## requirements
+
+The package uses the [Clang C++ library](https://clang.llvm.org/) and the [LibClang Python library](https://libclang.readthedocs.io/en/latest/index.html) to analyze the C++ code of a model. Both need to be installed and share the same version.
+
+The package provides an example script in [Example](/pycode/memilio-generation/memilio/tools/example_oseir.py). This script allows you to select which ODE model you want to bind. 
+For this package to use you need a model.cpp from the [Models folder](/cpp/models/..) and the corresponding file path.
+
+## Usage
+
+Before running the example script, configure the following settings:
+
+`conf.source_file`: Specifies the source file to be bound.
+
+`conf.target_folder`: Defines the folder where the generated Python file will be stored.
+
+
+Example Configuration:
+If you want to use the ode_secirvvs model the config should be like this -> `conf.source_file = home_and_user_folder + \ "/memilio/cpp/models/ode_secirvvs/model.cpp"`
+
+To use a different model, replace ode_secirvvs with the desired model's name in the file path.
+
+To set up a target folder, specify the desired output directory for the generated bindings.
+
+After setting up the source file and target folder, set the path to the example script in the terminal:
+If the terminal shows the memilio package, just give the path with `cd pycode/memilio-generation/memilio/tools`
+
+After the path is set, run the example script with the following command:
+
+```bash
+python example_oseir.py
+```
+
+## Visualization
+
+The package contains a [Visualization class](/pycode/memilio-generation/memilio/generation/graph_visualization.py) to display the generated AST.
+This allows for:
+
+Printing the AST in the terminal
+
+Saving it as a graph (PNG format)
+
+Formatting it in a text file
+
+You can print the AST in the example script with the aviz instance of the Visualisation() class.
+
+
+Example:
+`aviz.output_ast_formatted(ast, ast.get_node_by_index(1))` displays the second node of the AST and its children in a file called ast_formatted.txt. 
+The root node (`.get_node_by_index(0)`) displays the whole AST.
+
+`aviz.output_ast_terminal(ast, ast.get_node_by_index(1))` displays the second node of the AST and its children in terminal.
+
+The first argument of the statements specify the given AST. The second argument specifies the node and its children that you want to display.
+
+`aviz.output_ast_png(ast.get_node_by_index(2), 2)` displays the third node of the AST and its children with a depth of 2 as a png file. The second argument of the statement specifies the depth up to which the function displays child nodes. This means that any nodes beyond the specified depth (e.g., all nodes at level 3 and beyond if the depth is set to 2) will not be shown.
+
+To enable visualization, set the desired method under the `if (print_ast):` statement in the example script.
```

Comment:
Maybe rewrite to:
"The visualization is specified under the `if (print_ast):` statement in the example script. It is enabled when running the example with the following command: